charms/interface_external_cloud_provider.py

```diff
@@ -115,10 +118,9 @@ def metadata(self) -> Optional[dict]:
             metadata = metadata_req.read() if metadata_req.status == 200 else None
             return json.loads(metadata) if metadata else None
         elif self.name == "gce":
-            req = Request(AZURE_METADATA, headers={"Metadata": "true"})
+            req = Request(GOOGLE_METADATA, headers={"Metadata-Flavor": "Google"})
             metadata_req = urlopen(req)
-            metadata = metadata_req.read() if metadata_req.status == 200 else None
-            return json.loads(metadata) if metadata else None
+            return {"id": metadata_req.read()} if metadata_req.status == 200 else None
```

Comment:
I think this one returns bytes, the other provider return a json, which should be fine for loading.
```suggestion
            return {"id": metadata_req.read().decode()} if metadata_req.status == 200 else None
```